File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    first = [1, 2, 3, 4, 5]
    second = [2, 1, 2, 3, 2, 4, 2, 5]
    third = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    first_count = len(first)
    second_count = len(second)
    third_count = len(third)
    first_correct = 0
    second_coreect = 0
    third_correct = 0
    answer = []
    for i in range(0, len(answers)):
        if answers[i] == first[i % first_count]:
            first_correct += 1

        if answers[i] == second[i % second_count]:
            second_coreect += 1

        if answers[i] == third[i % third_count]:
            third_correct += 1
    max_correct = max(first_correct, second_coreect, third_correct)
    if max_correct == first_correct:
        answer.append(1)
    if max_correct == second_coreect:
        answer.append(2)
    if max_correct == third_correct:
        answer.append(3)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.swift
import Foundation

func solution(_ answers:[Int]) -> [Int] {
    var answerOne = [Bool]()
    var answerTwo = [Bool]()
    var answerThree = [Bool]()
    let first = [1,2,3,4,5]
    let second = [2,1,2,3,2,4,2,5]
    let third = [3,3,1,1,2,2,4,4,5,5]
    for i in 0..<answers.count {
        let answer = answers[i]
        
        if answer == first[i % first.count] {
            answerOne.append(true)
        }
        else {
            answerOne.append(false)
        }
        
        if answer == second[i % second.count] {
            answerTwo.append(true)
        }
        else {
            answerTwo.append(false)
        }
        
        if answer == third[i % third.count] {
            answerThree.append(true)
        }
        else {
            answerThree.append(false)
        }
        
    }
    var answer = [Int]()
    let count1 = answerOne.filter { $0 == true}.count
    let count2 = answerTwo.filter { $0 == true}.count
    let count3 = answerThree.filter { $0 == true}.count
    let count = [count1, count2, count3]
    let max = count.max()
    for i in 0..<count.count {
        if count[i] == max {
            answer.append(i+1)
        }
    }
    
    return answer
}